src/main/java/org/ballerinax/datamapper/diagnostic/DataMapperDiagnostic.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinax.datamapper.diagnostic;
+
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.Location;
+import io.ballerina.tools.diagnostics.properties.DiagnosticProperty;
+
+import java.util.List;
+
+/**
+ * Represents New DataMapperCompilerExtension Diagnostics.
+ *
+ * @since 2.0.0
+ */
+
+public class DataMapperDiagnostic extends Diagnostic {
+    private final Location location;
+    private final DiagnosticInfo diagnosticInfo;
+    private final String message;
+
+    public DataMapperDiagnostic(Location location, DiagnosticInfo diagnosticInfo, String message) {
+        this.location = location;
+        this.diagnosticInfo = diagnosticInfo;
+        this.message = message;
+    }
+
+    @Override
+    public Location location() {
+        return location;
+    }
+
+    @Override
+    public DiagnosticInfo diagnosticInfo() {
+        return diagnosticInfo;
+    }
+
+    @Override
+    public String message() {
+        return message;
+    }
+
+    @Override
+    public List<DiagnosticProperty<?>> properties() {
+        return null;
```

Comment:
Is there any reason for this to be `null` ?

Reply:
Made it null because there is nothing to return.